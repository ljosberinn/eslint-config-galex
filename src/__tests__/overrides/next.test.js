const { createNextJsOverride } = require('../../overrides/next');

describe('createNextJsOverride', () => {
  test('matches snapshot without next', () => {
    const project = {
      react: {
        hasReact: true,
        isNext: false,
      },
      typescript: {
        hasTypeScript: false,
      },
    };

    expect(createNextJsOverride(project)).toMatchSnapshot();
  });

  test('matches snapshot with next', () => {
    const project = {
      react: {
        hasReact: true,
        isNext: true,
      },
      typescript: {
        hasTypeScript: false,
      },
    };

    expect(createNextJsOverride(project)).toMatchSnapshot();
  });

  test('allows passing extra rules', () => {
    const rule = 'testing-library/no-manual-cleanup';
    const level = 'off';

    const project = {
      react: {
        hasReact: true,
        isNext: true,
      },
      rules: {
        [rule]: level,
      },
      typescript: {
        hasTypeScript: false,
      },
    };

    const override = createNextJsOverride({ ...project, rules: {} });

    const result = createNextJsOverride(project);

    expect(result.rules[rule]).toBe(level);
    expect(result.rules[rule]).not.toBe(override.rules[rule]);

    expect(result).toMatchSnapshot();
  });

  test('allows passing extra files', () => {
    const files = ['*.*.js'];

    const project = {
      react: {
        hasReact: true,
        isNext: true,
      },
      files,
      typescript: {
        hasTypeScript: false,
      },
    };

    const override = createNextJsOverride({ ...project, files: [] });

    const result = createNextJsOverride(project);

    expect(result.files).toContain(files[0]);
    expect(result.files).not.toBe(override.files);

    expect(result).toMatchSnapshot();
  });
});
