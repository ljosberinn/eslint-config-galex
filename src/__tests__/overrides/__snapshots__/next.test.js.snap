// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createNextJsOverride allows passing extra files 1`] = `
Object {
  "files": Array [
    "src/pages/*.?(js|ts)?(x)",
    "pages/*.?(js|ts)?(x)",
    "*.*.js",
  ],
  "rules": Object {
    "@next/next/missing-preload": "warn",
    "@next/next/no-css-tags": "warn",
    "@next/next/no-html-link-for-pages": "warn",
    "@next/next/no-sync-scripts": "warn",
    "@next/next/no-unwanted-polyfillio": "warn",
    "import/no-default-export": "off",
    "jsx-a11y/anchor-is-valid": Array [
      "error",
      Object {
        "aspects": Array [
          "invalidHref",
        ],
        "components": Array [
          "Link",
        ],
      },
    ],
    "react/react-in-jsx-scope": "off",
  },
}
`;

exports[`createNextJsOverride allows passing extra rules 1`] = `
Object {
  "files": Array [
    "src/pages/*.?(js|ts)?(x)",
    "pages/*.?(js|ts)?(x)",
  ],
  "rules": Object {
    "@next/next/missing-preload": "warn",
    "@next/next/no-css-tags": "warn",
    "@next/next/no-html-link-for-pages": "warn",
    "@next/next/no-sync-scripts": "warn",
    "@next/next/no-unwanted-polyfillio": "warn",
    "import/no-default-export": "off",
    "jsx-a11y/anchor-is-valid": Array [
      "error",
      Object {
        "aspects": Array [
          "invalidHref",
        ],
        "components": Array [
          "Link",
        ],
      },
    ],
    "react/react-in-jsx-scope": "off",
    "testing-library/no-manual-cleanup": "off",
  },
}
`;

exports[`createNextJsOverride matches snapshot with next 1`] = `
Object {
  "files": Array [
    "src/pages/*.?(js|ts)?(x)",
    "pages/*.?(js|ts)?(x)",
  ],
  "rules": Object {
    "@next/next/missing-preload": "warn",
    "@next/next/no-css-tags": "warn",
    "@next/next/no-html-link-for-pages": "warn",
    "@next/next/no-sync-scripts": "warn",
    "@next/next/no-unwanted-polyfillio": "warn",
    "import/no-default-export": "off",
    "jsx-a11y/anchor-is-valid": Array [
      "error",
      Object {
        "aspects": Array [
          "invalidHref",
        ],
        "components": Array [
          "Link",
        ],
      },
    ],
    "react/react-in-jsx-scope": "off",
  },
}
`;

exports[`createNextJsOverride matches snapshot without next 1`] = `null`;
